spring-cloud-generator/src/test/java/com/google/cloud/generator/spring/composer/goldens/EchoSpringAutoConfigurationRest.golden

```diff
@@ -0,0 +1,230 @@
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.cloud.spring.autoconfigure.core.GcpContextAutoConfiguration;
+import com.google.cloud.spring.core.Credentials;
+import com.google.cloud.spring.core.DefaultCredentialsProvider;
+import com.google.cloud.spring.core.Retry;
+import com.google.cloud.spring.core.util.RetryUtil;
+import com.google.showcase.v1beta1.EchoClient;
+import com.google.showcase.v1beta1.EchoSettings;
+import java.io.IOException;
+import java.util.Collections;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+// AUTO-GENERATED DOCUMENTATION AND CLASS.
+/**
+ * Auto-configuration for {@link EchoClient}.
+ *
+ * <p>Provides auto-configuration for Spring Boot
+ *
+ * <p>The default instance has everything set to sensible defaults:
+ *
+ * <ul>
+ *   <li>The default transport provider is used.
+ *   <li>Credentials are acquired automatically through Application Default Credentials.
+ *   <li>Retries are configured for idempotent methods but not for non-idempotent methods.
+ * </ul>
+ */
+@AutoConfiguration
+@AutoConfigureAfter(GcpContextAutoConfiguration.class)
+@ConditionalOnClass(EchoClient.class)
+@ConditionalOnProperty(value = "com.google.showcase.v1beta1.echo.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(EchoSpringProperties.class)
+public class EchoSpringAutoConfiguration {
+  private final EchoSpringProperties clientProperties;
+  private final CredentialsProvider credentialsProvider;
+  private static final Log LOGGER = LogFactory.getLog(EchoSpringAutoConfiguration.class);
+
+  protected EchoSpringAutoConfiguration(
+      EchoSpringProperties clientProperties, CredentialsProvider credentialsProvider)
+      throws IOException {
+    this.clientProperties = clientProperties;
+    if (this.clientProperties.getCredentials().hasKey()) {
+      if (LOGGER.isTraceEnabled()) {
+        LOGGER.trace("Using credentials from Echo-specific configuration");
+      }
+      this.credentialsProvider =
+          ((CredentialsProvider) new DefaultCredentialsProvider(this.clientProperties));
+    } else {
+      this.credentialsProvider = credentialsProvider;
+    }
+  }
+
+  /**
+   * Provides a default transport channel provider bean. The default is gRPC and will default to it
```

Comment:
I guess this javadoc is no longer correct for REST-only or gRPC-only services.

Reply:
Thanks for catching this! I missed updating the Javadoc generation logic here, will put in a fix.

Reply:
Updated javadoc content in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1849/commits/06b437cd44febaaa6592052bed9b643c4dfdb4e9.